From 8f31f2912eab8e0cfc2e0c7f4ff3d94fb4851a2a Mon Sep 17 00:00:00 2001
From: P Praneesh <quic_ppranees@quicinc.com>
Date: Mon, 3 Jan 2022 11:44:24 +0530
Subject: [PATCH 01/14] ath12k: remove monitor status and monitor desc rings

Remove unused monitor status and desc rings from driver, since
it monitor dest and monitor buffer rings handle the functionality.

Signed-off-by: P Praneesh <quic_ppranees@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/dp.c    | 14 ++++----------
 drivers/net/wireless/ath/ath12k/dp_tx.c |  8 --------
 drivers/net/wireless/ath/ath12k/hal.h   |  2 --
 3 files changed, 4 insertions(+), 20 deletions(-)

--- a/drivers/net/wireless/ath/ath12k/dp.c
+++ b/drivers/net/wireless/ath/ath12k/dp.c
@@ -266,6 +266,7 @@ int ath12k_dp_srng_setup(struct ath12k_b
 		params.intr_batch_cntr_thres_entries = 0;
 		params.intr_timer_thres_us = HAL_SRNG_INT_TIMER_THRESHOLD_RX;
 		break;
+	case HAL_TX_MONITOR_BUF:
 	case HAL_TX_MONITOR_DST:
 		params.low_threshold = DP_TX_MONITOR_BUF_SIZE_MAX >> 3;
 		params.flags |= HAL_SRNG_FLAGS_LOW_THRESH_INTR_EN;
@@ -292,7 +293,6 @@ int ath12k_dp_srng_setup(struct ath12k_b
 	case HAL_WBM_IDLE_LINK:
 	case HAL_SW2WBM_RELEASE:
 	case HAL_RXDMA_MONITOR_DST:
-	case HAL_RXDMA_MONITOR_DESC:
 		params.intr_batch_cntr_thres_entries =
 					HAL_SRNG_INT_BATCH_THRESHOLD_OTHER;
 		params.intr_timer_thres_us = HAL_SRNG_INT_TIMER_THRESHOLD_OTHER;
@@ -727,11 +727,8 @@ void ath12k_dp_link_desc_cleanup(struct
 				 u32 ring_type, struct dp_srng *ring)
 {
 	ath12k_dp_link_desc_bank_free(ab, desc_bank);
-
-	if (ring_type != HAL_RXDMA_MONITOR_DESC) {
-		ath12k_dp_srng_cleanup(ab, ring);
-		ath12k_dp_scatter_idle_link_desc_cleanup(ab);
-	}
+	ath12k_dp_srng_cleanup(ab, ring);
+	ath12k_dp_scatter_idle_link_desc_cleanup(ab);
 }
 
 static int ath12k_wbm_idle_ring_setup(struct ath12k_base *ab, u32 *n_link_desc)
@@ -813,8 +810,7 @@ int ath12k_dp_link_desc_setup(struct ath
 	tot_mem_sz = entry_sz * n_link_desc;
 
 	/* Setup scatter desc list when the total memory requirement is more */
-	if (tot_mem_sz > DP_LINK_DESC_ALLOC_SIZE_THRESH &&
-	    ring_type != HAL_RXDMA_MONITOR_DESC) {
+	if (tot_mem_sz > DP_LINK_DESC_ALLOC_SIZE_THRESH) {
 		ret = ath12k_dp_scatter_idle_link_desc_setup(ab, tot_mem_sz,
 							     n_link_desc_bank,
 							     n_link_desc,
--- a/drivers/net/wireless/ath/ath12k/dp_tx.c
+++ b/drivers/net/wireless/ath/ath12k/dp_tx.c
@@ -720,10 +720,6 @@ ath12k_dp_tx_get_ring_id_type(struct ath
 		*htt_ring_id = HTT_RXDMA_MONITOR_DEST_RING;
 		*htt_ring_type = HTT_HW_TO_SW_RING;
 		break;
-	case HAL_RXDMA_MONITOR_DESC:
-		*htt_ring_id = HTT_RXDMA_MONITOR_DESC_RING;
-		*htt_ring_type = HTT_SW_TO_HW_RING;
-		break;
 	case HAL_TX_MONITOR_BUF:
 		*htt_ring_id = HTT_TX_MON_HOST2MON_BUF_RING;
 		*htt_ring_type = HTT_SW_TO_HW_RING;
--- a/drivers/net/wireless/ath/ath12k/hal.h
+++ b/drivers/net/wireless/ath/ath12k/hal.h
@@ -458,8 +458,8 @@ enum hal_srng_ring_id {
 	HAL_SRNG_RING_ID_WMAC1_SW2RXDMA0_BUF = HAL_SRNG_RING_ID_PMAC1_ID_START,
 	HAL_SRNG_RING_ID_WMAC1_SW2RXDMA1_BUF,
 	HAL_SRNG_RING_ID_WMAC1_SW2RXDMA2_BUF,
-	HAL_SRNG_RING_ID_WMAC1_SW2RXMON_BUF0,
-
+	HAL_SRNG_RING_ID_WMAC1_SW2RXDMA0_STATBUF,
+	HAL_SRNG_RING_ID_WMAC1_SW2RXDMA1_STATBUF,
 	HAL_SRNG_RING_ID_WMAC1_RXDMA2SW0,
 	HAL_SRNG_RING_ID_WMAC1_RXDMA2SW1,
 	HAL_SRNG_RING_ID_WMAC1_RXMON2SW0 = HAL_SRNG_RING_ID_WMAC1_RXDMA2SW1,
@@ -504,9 +504,8 @@ enum hal_ring_type {
 	HAL_RXDMA_BUF,
 	HAL_RXDMA_DST,
 	HAL_RXDMA_MONITOR_BUF,
-	HAL_RXDMA_MONITOR_STATUS,
 	HAL_RXDMA_MONITOR_DST,
-	HAL_RXDMA_MONITOR_DESC,
+	HAL_RXDMA_MONITOR_STATUS,
 	HAL_RXDMA_DIR_BUF,
 	HAL_PPE2TCL,
 	HAL_PPE_RELEASE,
--- a/drivers/net/wireless/ath/ath12k/hal.c
+++ b/drivers/net/wireless/ath/ath12k/hal.c
@@ -144,7 +144,14 @@ static const struct hal_srng_config hw_s
 		.ring_dir = HAL_SRNG_DIR_DST,
 		.max_size = HAL_RXDMA_RING_MAX_SIZE_BE,
 	},
-
+	{ /* RXDMA_MONITOR_STATUS */
+		.start_ring_id = HAL_SRNG_RING_ID_WMAC1_SW2RXDMA1_STATBUF,
+		.max_rings = 1,
+		.entry_size = sizeof(struct hal_wbm_buffer_ring) >> 2,
+		.mac_type = ATH12K_HAL_SRNG_PMAC,
+		.ring_dir = HAL_SRNG_DIR_SRC,
+		.max_size = HAL_RXDMA_RING_MAX_SIZE,
+	},
 	{ /* RXDMA DIR BUF */
 		.start_ring_id = HAL_SRNG_RING_ID_RXDMA_DIR_BUF,
 		.max_rings = 2,
@@ -184,7 +191,7 @@ static const struct hal_srng_config hw_s
 		.max_size = HAL_RXDMA_RING_MAX_SIZE_BE,
 	},
 	{ /* RXDMA_MONITOR_DEST */
-		.start_ring_id = HAL_SRNG_RING_ID_WMAC1_SW2RXMON_BUF0,
+		.start_ring_id = HAL_SRNG_RING_ID_WMAC1_RXMON2SW0,
 		.max_rings = 1,
 		.entry_size = sizeof(struct hal_mon_dest_desc) >> 2,
 		.ring_dir = HAL_SRNG_DIR_DST,
